c_src/sqlite3_nif.c

```diff
@@ -182,20 +183,12 @@ make_bind_error(ErlNifEnv* env, ERL_NIF_TERM message, ERL_NIF_TERM argument)
 }
 
 static ERL_NIF_TERM
-make_binary(ErlNifEnv* env, const void* bytes, unsigned int size)
+make_binary(ErlNifEnv* env, const char* bytes, size_t size)
 {
-    ErlNifBinary blob;
-    ERL_NIF_TERM term;
-
-    if (!enif_alloc_binary(size, &blob)) {
-        return make_atom(env, "out_of_memory");
-    }
-
-    memcpy(blob.data, bytes, size);
-    term = enif_make_binary(env, &blob);
-    enif_release_binary(&blob);
-
-    return term;
+    ERL_NIF_TERM bin;
+    const char* data = enif_make_new_binary(env, size, &bin);
```

Comment:
I think I didn't use this function because of the documentation: https://www.erlang.org/doc/apps/erts/erl_nif.html#enif_make_new_binary

> The drawbacks are that the binary cannot be kept between NIF calls and it cannot be reallocated.

Is this saying that we shouldn't store it in a global `static ERL_NIF_TERM` and that it should be safe to pass back to the VM but don't rely on it being present?

Reply:
Yes, I think that's what it means. I'm pretty sure it's safe to use since Exqlite only keeps conn and stmt resources between nif calls.